src/Module.cpp

```diff
@@ -534,10 +534,21 @@ void Module::compile(const std::map<OutputFileType, std::string> &output_files)
         Internal::print_to_html(output_files.at(OutputFileType::stmt_html), *this);
     }
 
+    // Minor but worthwhile optimization: if all of the output files are of types that won't
+    // ever rely on submodules (e.g.: toplevel declarations in C/C++), don't bother resolving
+    // the submodules, which can call compile_to_buffer().
+    const auto should_ignore_submodules = [](const std::map<OutputFileType, std::string> &output_files) {
+        auto output_files_copy = output_files;
```

Comment:
A non-mutating way to do this would be `output_files.size() == output_files.count(...) + output_files.count(...) + output_files.count(...)`

Reply:
Lazy programmer was lazy, done